### PR TITLE
feat: add script for writing results to README and checkout ref

### PR DESCRIPTION
**Pull request by genryxy**
Part of artipie/artipie#431
Added script for writing results to README and made checkout by specified ref

### CODE REVIEW

**Review by olenagerasimova — DISMISSED** [review]
@genryxy thanks, though I'd suggest asking Kirill to review this PR

**Comment by genryxy** [comment]
@g4s8 could you please take a look

**Comment by g4s8** [comment]
@genryxy where this script will add results in README file?

**Comment by genryxy** [comment]
@g4s8 in [these ](https://github.com/artipie/benchmarks/pull/38/files#diff-66c3d19864f5e0aa28635807694445eb6af1729c963f3364e21972c221a370bdR107-R113) lines new results are added to the README

**Review by g4s8 — CHANGES_REQUESTED** [review]
@genryxy so what happens if the readme is not ended with benchmarks table? It seems this script will add invalid markdown syntax to the end of readme page in that case

**Comment by genryxy** [comment]
@g4s8 it will change the content of the file. More precisely, it will move the entire text below the table above the table. If there are several tables in this README file, it won't work correctly

**Comment by genryxy** [comment]
@g4s8 here some examples

**First (new bench)** 
--
The table below contains info about results of benchmarks for different versions for existed SDK API operations.

| Version | HelmAstoAdd | HelmAstoRemove | HelmAstoReindex |
|:---:|:---:|:---:|:---:|
| 0.3 | 50.708 ± 0.115 | 51.498 ± 0.165 | 6.858 ± 0.009 |
| 1.0.1 | 50.454 ± 0.111 | 51.308 ± 0.173 | 8.015 ± 0.017 |  
| 1.1.1 | 30.448 ± 0.152 | 31.055 ± 0.096 | 5.342 ± 0.011 |

**Result**
--
The table below contains info about results of benchmarks for different versions for existed SDK API operations.

| Version | HelmAstoAdd | HelmAstoRemove | HelmAstoReindex | newBench |
|:---:|:---:|:---:|:---:|:---:|
| 0.3 | 50.708 ± 0.115 | 51.498 ± 0.165 | 6.858 ± 0.009 |-|
| 1.0.1 | 50.454 ± 0.111 | 51.308 ± 0.173 | 8.015 ± 0.017 |-|  
| 1.1.1 | 30.448 ± 0.152 | 31.055 ± 0.096 | 5.342 ± 0.011 |-|
|new1.1.2|val1|val2|val3|val4|

**Second (old bench did not run)**
--
Some text here.

| Version | HelmAstoAdd | HelmAstoRemove | HelmAstoReindex |
|:---:|:---:|:---:|:---:|
| 0.3 | 50.708 ± 0.115 | 51.498 ± 0.165 | 6.858 ± 0.009 |
| 1.0.1 | 50.454 ± 0.111 | 51.308 ± 0.173 | 8.015 ± 0.017 |  
| 1.1.1 | 30.448 ± 0.152 | 31.055 ± 0.096 | 5.342 ± 0.011 |

**Result**
--
Some text here.

| Version | HelmAstoAdd | HelmAstoRemove | HelmAstoReindex | 
|:---:|:---:|:---:|:---:|
| 0.3 | 50.708 ± 0.115 | 51.498 ± 0.165 | 6.858 ± 0.009 |
| 1.0.1 | 50.454 ± 0.111 | 51.308 ± 0.173 | 8.015 ± 0.017 |
| 1.1.1 | 30.448 ± 0.152 | 31.055 ± 0.096 | 5.342 ± 0.011 |
|new1.1.2|val1|val2|-|

**Third (without table)**
--
Some text here. Here is not any table.

**Result**
--
Some text here. Here is not any table.

| Version | HelmAstoAdd | HelmAstoRemove | HelmAstoReindex | 
|:---:|:---:|:---:|:---:|
| 0.3 | 50.708 ± 0.115 | 51.498 ± 0.165 | 6.858 ± 0.009 |

**Fourth (text below the table)**
--
Some text here above the table.
| Version | HelmAstoAdd | HelmAstoRemove | HelmAstoReindex | 
|:---:|:---:|:---:|:---:|
| 0.3 | 50.708 ± 0.115 | 51.498 ± 0.165 | 6.858 ± 0.009 |
| 1.0.1 | 50.454 ± 0.111 | 51.308 ± 0.173 | 8.015 ± 0.017 |

**Some text here below the table.**

**Result**
--
Some text here above the table.
**Some text here below the table.**
| Version | HelmAstoAdd | HelmAstoRemove | HelmAstoReindex | 
|:---:|:---:|:---:|:---:|
| 0.3 | 50.708 ± 0.115 | 51.498 ± 0.165 | 6.858 ± 0.009 |
| 1.0.1 | 50.454 ± 0.111 | 51.308 ± 0.173 | 8.015 ± 0.017 |
|new1.1.2|val1|val2|val3|

**Fifth (several tables in README)**
--
Some text here above the table.
| Version | HelmAstoAdd | HelmAstoRemove | HelmAstoReindex | 
|:---:|:---:|:---:|:---:|
| 0.3 | 50.708 ± 0.115 | 51.498 ± 0.165 | 6.858 ± 0.009 |
| 1.0.1 | 50.454 ± 0.111 | 51.308 ± 0.173 | 8.015 ± 0.017 |

|Another|table|
|:---:|:---:|
|it is|unexpected|

**Result**
--
The result will be wrong because also values from the second table will be taken.

**Comment by g4s8** [comment]
@genryxy I see, maybe we can update another file specified by user, not README?

**Comment by genryxy** [comment]
@g4s8 thanks, now it will be written to a separate file `results.md`. And link to this file could be added to the readme file

**Review by g4s8 — DISMISSED** [review]
@genryxy thanks

**Comment by genryxy** [comment]
@g4s8 could you please merge

**Comment by genryxy** [comment]
@g4s8 please reapprove because I moved script to a separate folder and renamed file with it

**Review by g4s8 — CHANGES_REQUESTED** [review]
@genryxy sorry, it seems I didn't understand the what is the output file for these results. Now it looks like it was `./benchmarks/README.md`, am I right? Then it looks like a correct place for these results. In any case it'd be better to specify it as input parameter for this script: `python3 ./writeresults.py ./results/README.md` for more flexible configuration and debugging. Also, I don't see how this script will be executed, I think we need to add it as a Makefile target.

**Comment by genryxy** [comment]
@g4s8 I supposed to add file `./benchmarks/results.md` which would contain only table with results. And specify a link to this table in `./benchmarks/README.md`

**Comment by genryxy** [comment]
@g4s8 thanks, made as you suggested

File: adapters/Makefile [hunk 1]
```diff
@@ -27,3 +29,10 @@ prepare: generate
 
 run: generate build prepare
 		/bin/bash ./script.sh run $(TARGET)
+
+write_results:
```
Comment (g4s8) [inline adapters/Makefile]: Does this target depend on another? As I understood, it requires some data to generate this report. Then we need a target for this data which depends on `run`: `write_results -> benchmark_data -> run`, where `benchmark_data` can process result data produced by `run` target

**Review by g4s8 — CHANGES_REQUESTED** [review]
@genryxy please check my comment

**Comment by genryxy** [comment]
@g4s8 thanks, corrected. I thought at first call `run` and after that on another step call `draw_charts` stage because it looked for me like two different operations. But yes, it would be better join these stages into one to avoid duplicating code for calling these two stages.

**Comment by genryxy** [comment]
@g4s8 could you please take a look

**Comment by genryxy** [comment]
@g4s8 could you please check

**Comment by genryxy** [comment]
@g4s8 please check

**Review by g4s8 — COMMENTED** [review]
@genryxy thanks

**Review by g4s8 — APPROVED** [review]

**Pull request merged**